### DISCUSSION TITLE
[Bug]: Search result with broken HTML for matches inside links

### DISCUSSION THREAD

**Discussion by hermannromanek** [django-wiki/django-wiki#1302] [Bug]
### Describe the bug

When searched terms are found within a links slug, the filter get_content_snippet returns invalid HTML.

Problem is the filter cleaning HTML (maybe intentionally to also search through that?) for the keyword. My quick fix was to do a strip_tags at the start of the function:

![grafik](https://github.com/django-wiki/django-wiki/assets/129487986/6ffe96fd-e828-4bc1-8bd9-1df18f71645c)

This probably means it is no longer neccessary in the nested function clean_text.

### Steps to reproduce.

1. Create a page under slug lorem-ipsum
2. Create a second page linking to this, e.g. with content:
   `The text [Lorem Ipsum](/lorem-ipsum/) will display incorrectly when searching for "lorem".`
3. On a scope containing the buggy page, search for term "lorem".

![grafik](https://github.com/django-wiki/django-wiki/assets/129487986/114bcb46-f3a9-4163-8a36-97685d2ed352)

### Expected behavior

Expected would be correct HTML, highlighting the search term within the a tag.

May require more complex logic if the search should find slugs that are named differently than the link pointing there.

### Python Version

3.9+

### Django Version

4.0+

**Comment by benjaoming** [comment]
Thanks for the report :pray: 

Would it be correct to assume that the intention is to be able to find the search keyword if it's part of some HTML attribute? For instance URL slugs or image ALT texts?

**Reply by hermannromanek** [reply]
Thanks for the speedy reply :)


In case of my proposed fix its the opposite - the highlighting disregards any HTML tags and attributes since those can not be highlighted anyway, in order to produce valid HTML and not break the highlighting on other parts. I think it is a perfectly viable way to approach this, working on the assumption the searched term will occur both in the slug as well as in the text itself.

Highlighting something because of an attribute could be nice, but:
- those cases may be rather complex and would probably require parsing the HTML
- it may be confusing for the reader, searching for "foo" and having the term "bar" highlighted because of something non-obvious like an alt or title attribute containing "foo"
- the results will be found and listed either way, we're just talking display here. So in the edge case that the searched text is found only in an attribute, nothing is highlighted but the page is still included in the search results, just without any highlighting.

In my case and considering especially the last of the mentioned points - display only, no results lost - I thought the benefits of a more extensive solution would not justify the required effort and complexity.

**Reply by benjaoming** [reply]
I don't have any strong opinions here :) I can't see any test cases addressing this, so I don't think anyone has targeted the behavior previously.

I think it sounds better to have "nothing" displayed rather than "broken HTML" since the latter could indicate that the wiki is somehow broken :+1: 

But solving it should probably come with a real test case, so we can maintain the behavior in the future :+1:

**Comment by benjaoming** [comment]
Moved to an issue :+1: 

https://github.com/django-wiki/django-wiki/issues/1303
